File: C++/Sqrt(x).cpp
class Solution {
public:
    int mySqrt(int x) {
        if (x <= 1) return x;
        long long low = 1, high = x, res = x;
        while (low <= high) {
            long long mid = low + (high - low) / 2;
            if (mid * mid <= res && (mid + 1) * (mid + 1) > res) return mid;
            else if (mid * mid > res) high = mid - 1;
            else low = mid + 1;
        }
    }
};

File: Python/Sqrt(x).py
#!/usr/bin/env python

class Solution:
    # @param x, an integer
    # @return an integer
    def sqrt(self, x):
        left = 0
        right = x
        while left <= right:
            mid = (left + right) / 2
            if mid * mid <= x and (mid + 1) * (mid + 1) > x:
                return mid
            elif mid * mid > x:
                right = mid - 1
            else:
                left = mid + 1